tests/unit/challenges/test_aws_utils.py

```diff
@@ -0,0 +1,214 @@
+            title="Test Challenge 2",
+            creator=self.challenge_host_team,
+            queue="test_queue_2",
+            start_date=timezone.now() - timedelta(days=1),
+            end_date=timezone.now() + timedelta(days=1),
+        )
+        with self.settings(MEDIA_ROOT="/tmp/evalai"):
+            self.challenge_phase = ChallengePhase.objects.create(
+                name="Challenge Phase",
+                description="Description for Challenge Phase",
+                start_date=timezone.now() - timedelta(days=1),
+                end_date=timezone.now() + timedelta(days=1),
+                challenge=self.challenge,
+            )
+            self.challenge_phase2 = ChallengePhase.objects.create(
+                name="Challenge Phase 2",
+                description="Description for Challenge Phase",
+                start_date=timezone.now() - timedelta(days=1),
+                end_date=timezone.now() + timedelta(days=1),
+                challenge=self.challenge2,
+            )
+
+        self.client.force_authenticate(user=self.user)
+        self.ecs_client = boto3.client("ecs", region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"), aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"), aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),)
+
+
+@mock_ecs
+class BaseAdminCallClass(BaseTestClass):
+    def setUp(self):
+        aws_utils.COMMON_SETTINGS_DICT["CLUSTER"] = "testcluster"
+        super(BaseAdminCallClass, self).setUp()
+
+        self.challenge3 = Challenge.objects.create(
+            title="Test Challenge 3",
+            creator=self.challenge_host_team,
+            queue="test_queue_3",
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+        )
+        self.ecs_client.create_cluster(clusterName="testcluster")
+        self.client_token = "abi481"
+
+    @classmethod
+    def queryset(cls, pklist):
+        queryset = Challenge.objects.filter(pk__in=pklist)
+        queryset = sorted(queryset, key=lambda i: pklist.index(i.pk))
+        return queryset
+
+
+class TestDeleteWorkers(BaseAdminCallClass):
+    def setUp(self):
+        super(TestDeleteWorkers, self).setUp()
+
+    def test_delete_workers_when_all_three_challenges_have_active_workers(self):
+        aws_utils.create_service_by_challenge_pk(
+            self.ecs_client,
+            self.challenge,
+            self.client_token)
+        aws_utils.create_service_by_challenge_pk(
+            self.ecs_client,
+            self.challenge2,
+            self.client_token)
+        aws_utils.create_service_by_challenge_pk(
+            self.ecs_client,
+            self.challenge3,
+            self.client_token)
+
+        pk_list = [self.challenge.pk, self.challenge2.pk, self.challenge3.pk]
+        queryset = super(TestDeleteWorkers, self).queryset(pk_list)
+
+        count = 3
+        expected_workers = [None, None, None]
+        failures = []
```

Comment:
Please change the variable name from `failures` to `expected_failures`.  That would be more clear.